### ISSUE TITLE
[Cell Analysis] Custom thresholding with Canny Edge detection

### ISSUE THREAD

**Issue by TiraFire** [Data-Boom/SOEN490#388]
As a user, I would like to isolate the edges of the cells within the input image using Canny Edge detection and by manipulating the algorithm's following parameters: sigma, low and high thresholds. 

**Description** 
We are using canny edge detection from OpenCV in order to isolate edges in the image. Given that the input images are different from each others in terms of scale and quality of pictures, the edge detection algorithm needs to be fitted for each images. This is done by manipulating the sigma, low and high thresholds. We may implement a custom Canny Edge algorithm in order to do accomplish this. 

**Additional Notes**

**Tasks**
- [ ] Implement custom Canny Edge detection algorithm that can isolate edges
- [ ] Isolate edges for different cell images by manipulating the sigma, low and high thresholds (can be hard-coded)

**Acceptance Criteria**
- The edges of the cells are precisely visible due to manipulation of sigma, low and high threshold
- The edges are white and the rest of the image is black
- Various images are detected with their "best results"

**Comment by JeremiahTee** [comment]
In the first week and a half of Sprint 8, we implemented Canny Edge detection from scratch in order to identify and isolate the edges of cell sizes on foil images given to us by the stakeholder.
We found that we would get very different results for each image, since the scanned images vary in intensity of pixels and the edges are not clearly shown. We needed to set different thresholds to identify edges for each image.
There is no universal edge detection algorithm that works for all images, without the requirement of machine learning.
Since machine learning is out of the scope for the remainder project (as agreed with our stakeholder), we came up with a solution that has been approved by our stakeholder.

The solution requires human interaction where arbitrary lines are drawn on the image, and then through computer vision / mathematical computation we extrapolate the rest of the lines,
This solution was proposed and approved by the stakeholder on Thursday February 18th, and so we did not have the time to implement it for Sprint 7. 

We are putting this issue on the icebox pipeline, and we will update this issue for Sprint 8 and re-estimate it.
